components/GroupView/GroupView.js

```diff
@@ -33,36 +35,47 @@ export default class GroupView extends Component {
     updateUserLocation();
   }
 
-  _handleChangePage() {
+  _handleChangePage(name) {
     this.props.navigator.push({
       component: GroupMapChat,
-      title: 'GroupNameGoesHere',
+      title: name + ' Group',
       passProps: {
         user: this.props.user,
-      },
+        groupName: name,
+      }
     });
-  }
+  };
 
   _usersListener() {
-    const context = this;
-    this.usersRef.on('value', (snapshot) => {
-      const usersArray = [];
-      snapshot.forEach((childSnapshot) => {
+    this.usersRef.on('value', function(snapshot) {
+      let usersArray = [];
+      snapshot.forEach(function(childSnapshot) {
         usersArray.push(childSnapshot.val());
-      });
-      context.setState({ users: usersArray });
-    });
+      })
+      this.setState({ users: usersArray});
+    }.bind(this))
   }
 
   render() {
-    const userList = this.state.users.map(user =>
-      (
-        <View style={styles.li} key={user.uid}>
+
+    const userList = this.state.users.map((user, i) => {
+      return (
+        <View style={styles.li} key={i}>
           <Text>{user.displayName}</Text>
-          <Text>Location: {user.location ? user.location.coords.longitude : 'null'}, {user.location ? user.location.coords.latitude : 'null'}</Text>
```

Comment:
This was to handle locations being undefined at creation of a new user. Probably still need this code? Although, if users aren't being listed here, it shouldn't matter that much because this will get deleted in the future.